src/wizard/obtain_source.jl

```diff
@@ -357,6 +357,49 @@ function get_name_and_version(state::WizardState)
     end
 end
 
+@enum Compilers C=1 Go Rust
+function get_compilers(state::WizardState)
+    while state.compilers === nothing
+        compiler_descriptions = Dict(C => "C/C++/Fortran", Go => "Go", Rust => "Rust")
+        compiler_symbols = Dict(Int(C) => :c, Int(Go) => :go, Int(Rust) => :rust)
+        terminal = TTYTerminal("xterm", state.ins, state.outs, state.outs)
+        result = nothing
+        while true
+            select_menu = MultiSelectMenu([compiler_descriptions[i] for i in instances(Compilers)])
+            select_menu.selected = Set([Int(C)])
+            result = request(terminal,
+                             "Select compilers for the project",
+                             select_menu
+                             )
+            if isempty(result)
+                println("Must select at least one platform")
+            else
+                break
+            end
+        end
+        state.compilers = map(c -> compiler_symbols[c], collect(result))
+    end
+end
+
+function get_preferred_version(state::WizardState, compiler::AbstractString,
+                               available_versions=Vector{Integer})
+    version = nothing
+    minver, maxver = getfield.(extrema(available_versions), :major)
+    while version === nothing #&& !eof(state.ins)
+        msg = "Please enter the preferred $(compiler) version ($(minver)...$(maxver))"
+        version = tryparse(Int, nonempty_line_prompt("$(compiler) version", msg;
+                                                     ins=state.ins, outs=state.outs))
+        if !(isnothing(version) || minver <= version <= maxver)
```

Comment:
I'm not sure this is necessary; we'll always choose the closest version, so if someone asks for GCC `v"1000"`, we'll just give them `v"9"`.  While that would raise some eyebrows, I'm not sure it's necessary to disallow it in the wizard; better to be gracious, IMO.

Reply:
This is a temporary implementation: I'd like to use `RadioMenu`, but in order to do that I need to make it work in tests (that's why I asked about simulating arrow keys in tests), but I'll go back to this in the next days.